### PR TITLE
Make Adaguc call async.

### PR DESCRIPTION
**Pull request by lukas-phaf**
Also fix the postgres volume location in the docker compose file.

### CODE REVIEW

File: Docker/docker-compose.yml [hunk 1]
```diff
@@ -20,7 +20,8 @@ services:
         max-size: "200k"
         max-file: "10"
   adaguc-server:
-    image: openearth/adaguc-server
+#    image: openearth/adaguc-server
+    build: ..
```
Comment (maartenplieger) [inline Docker/docker-compose.yml]: Do we want this already in? I guess should leave it as is.

File: python/lib/adaguc/CGIRunner.py [hunk 2]
```diff
@@ -15,14 +16,18 @@
 import re
 #from numba import jit
 
+# TODO: This limits the number of adaguc process calls per fastapi worker.
+#   Do we want this?
+#   Make configurable
+sem = asyncio.Semaphore(4)
```
Comment (maartenplieger) [inline python/lib/adaguc/CGIRunner.py]: Can we make this configurable via an environment var or something?

Comment (maartenplieger) [reply python/lib/adaguc/CGIRunner.py]: Also in Docker/start.sh, the number of gunicorn workers is still 4.

Could this now be set to 1?

File: python/lib/adaguc/CGIRunner.py [hunk 3]
```diff
@@ -13,40 +14,48 @@
 import chardet
 from queue import Queue, Empty  # python 3.x
 import re
-#from numba import jit
 
+# from numba import jit
+
+ADAGUC_NUMPARALLELPROCESSES = os.getenv("ADAGUC_NUMPARALLELPROCESSES", "4")
+# print("ADAGUC_NUMPARALLELPROCESSES", ADAGUC_NUMPARALLELPROCESSES)
+sem = asyncio.Semaphore(int(ADAGUC_NUMPARALLELPROCESSES))
```
Comment (mgrunbauer) [inline python/lib/adaguc/CGIRunner.py]: Remove these comments please?

File: Docker/docker-compose-generate-env.sh [hunk 4]
```diff
@@ -5,11 +5,12 @@ ADAGUC_PORT=443
 ADAGUC_DATA_DIR=${HOME}/adaguc-docker/adaguc-data
 ADAGUC_AUTOWMS_DIR=${HOME}/adaguc-docker/adaguc-autowms
 ADAGUC_DATASET_DIR=${HOME}/adaguc-docker/adaguc-datasets
+ADAGUC_NUMPARALLELPROCESSES=8
```
Comment (mgrunbauer) [inline Docker/docker-compose-generate-env.sh]: Please set this to `4`

File: python/python_fastapi_server/routers/edr.py [hunk 5]
```diff
@@ -259,8 +259,8 @@ async def get_collection_position(
 }
 
 
-@cached(cache=TTLCache(maxsize=1024, ttl=60))
-def get_collectioninfo_for_id(
+# @cached(cache=TTLCache(maxsize=1024, ttl=60))
```
Comment (mgrunbauer) [inline python/python_fastapi_server/routers/edr.py]: Remove this comment?

File: python/python_fastapi_server/routers/edr.py [hunk 6]
```diff
@@ -612,12 +612,12 @@ async def rest_get_edr_collection_by_id(collection_name: str):
...
     GET Returns collection information for given collection id
     """
-    collection = get_collectioninfo_for_id(collection_name)
+    collection = await get_collectioninfo_for_id(collection_name)
     return collection
 
 
-@cached(cache=TTLCache(maxsize=1024, ttl=60))
-def get_capabilities(collname):
+# @cached(cache=TTLCache(maxsize=1024, ttl=60))
```
Comment (mgrunbauer) [inline python/python_fastapi_server/routers/edr.py]: Remove comments?

File: python/python_fastapi_server/routers/ogcapi_tools.py [hunk 7]
```diff
@@ -171,12 +171,12 @@ def get_dimensions(layer, skip_dims=None):
     return dims
 
 
-@cached(cache=cache)
-def get_parameters(collname):
+# @cached(cache=cache)
```
Comment (mgrunbauer) [inline python/python_fastapi_server/routers/ogcapi_tools.py]: Remove comments?

**Review by mgrunbauer — DISMISSED** [review]

File: python/lib/adaguc/CGIRunner.py [hunk 8]
```diff
@@ -13,40 +14,48 @@
...
-            output.write(b'TimeOut')
-            return 1, [], None
+        ON_POSIX = "posix" in sys.builtin_module_names
+        async with sem:
+            process = await asyncio.create_subprocess_exec(
+                *cmds, stdout=PIPE, stderr=PIPE, env=localenv, close_fds=ON_POSIX
+            )
+
+            try:
+                asyncio.wait_for(process, timeout=timeout)
```
Comment (maartenplieger) [inline python/lib/adaguc/CGIRunner.py]: It seems that that the timeout is not working. When I place an infinite loop in adguc-server, the process will hang.

**Review by mgrunbauer — APPROVED** [review]
🎉

**Pull request merged**